jetty-util/src/main/java/org/eclipse/jetty/util/AbstractTrie.java

```diff
@@ -81,4 +82,57 @@ public boolean isCaseInsensitive()
     {
         return _caseInsensitive;
     }
+
+    public static <T> Trie<T> emptyTrie(boolean caseInsensitive)
+    {
+        return new AbstractTrie<T>(caseInsensitive)
```

Comment:
If it's empty, should be irrelevant if it's case sensitive, so perhaps we can make this anonymous class a `static final`?
Method `isCaseSensitive()` is only used internally by the implementation (e.g. by `put(...)`, and since this is the empty implementation it should not matter.
Can it be moved to the interface, something like `Trie.EMPTY`?

Reply:
A constant like `Trie.EMPTY` does not work because it is a generic type, so you need a method.  Thus may as well have the caseinsensitive field.   But I've moved to the interface.